metrics.nim

```diff
@@ -102,11 +116,11 @@ when defined(metrics):
     labelRegexStr {.global.} = r"^[a-zA-Z_][a-zA-Z0-9_]*$"
     labelRegex {.global.} = re(labelRegexStr)
 
-  proc validateName(name: string) =
+  proc validateName(name: string) {.raises: [Defect, ValueError].} =
```

Comment:
Why add "Defect" in individual proc pragmas if you already added it with a "push" at the file level?

Reply:
because the local raises overrides the pushed version